### ISSUE TITLE
make: *** [Makefile:29: libs] Error 127

### ISSUE THREAD

**Issue by taoufikelka** [jdah/minecraft-weekend#50]
I'm getting this while building the code (Trying it on Linux 5.8.0-55-generic #62~20.04.1-Ubuntu, gcc 9.3.0)

make libs
cd lib/cglm && cmake . -DCGLM_STATIC=ON && make
-- Configuring done
-- Generating done
...
[ 90%] Building C object CMakeFiles/cglm.dir/src/ray.c.o
[ 95%] Building C object CMakeFiles/cglm.dir/src/affine2d.c.o
[100%] Linking C static library libcglm.a
make[3]: Leaving directory '/home/agentt/Divers/Programming/C/minecraft-weekend/lib/cglm'
[100%] Built target cglm
make[2]: Leaving directory '/home/agentt/Divers/Programming/C/minecraft-weekend/lib/cglm'
make[1]: Leaving directory '/home/agentt/Divers/Programming/C/minecraft-weekend/lib/cglm'
cd lib/glad && clang -o src/glad.o -Iinclude -c src/glad.c
/bin/sh: 1: clang: not found
make: *** [Makefile:29: libs] Error 127

**Comment by crims0nHD** [comment]
Seems like you don't have the clang compiler installed.

**Comment by TheTechRobo** [comment]
Yeah, you need to install the clang compiler. If it already is installed that's a different story but from the looks of things it isn't.

**Comment by taoufikelka** [comment]
> Seems like you don't have the clang compiler installed.

ow  yeah didn't see that line, thank you so much



> Yeah, you need to install the clang compiler. If it already is installed that's a different story but from the looks of things it isn't.

I don't think it's already installed, as I said before I'm on Ubuntu.
another question, as a beginner, is there a way around to compile using gcc, or I will have so many problems trying something like this ?

**Comment by crims0nHD** [comment]
Yes you could theoretically change the compiler used (I think you just have to change a variable like CC in the Makefile).
However, you also would need to adjust the compiler flags because gcc may have different naming conventions for some flags.
Last but not least but if you're unlucky you have to dig into the code to fix stuff that relies on the clang compiler.

TLDR: I would not suggest doing it, however if you have the time and patience, experiment around and try it, maybe even fork the project and let other people have a go at it. :)

**Comment by taoufikelka** [comment]
First of all thank you all for your time and answers.
This sounds like a new challenge, I will try it, and as you said @crims0nHD if I succeeded I will have a fork of the project for others.
